core/src/main/java/org/opentosca/toscana/core/csar/CsarService.java

```diff
@@ -0,0 +1,27 @@
+package org.opentosca.toscana.core.csar;
+
+import java.io.InputStream;
+import java.util.Collection;
+
+public interface CsarService {
+
+
+    /**
+     * Creates a new Csar instance
+     * @param csarStream the actual cloud service archive as InputStream
+     * @return the newly created Csar instance
+     */
+    Csar uploadCsar(InputStream csarStream);
```

Comment:
The name of the archive has to be specified here. Reading it from the archive file is very diffcult if not impossible. 

The server will not know the original file name before uploading!

Reply:
Alright. So now it's the clients responsibility to declare a name when creating a csar